src/Control/Lens/Iso.hs

```diff
@@ -409,6 +415,57 @@ instance Swapped (,) where
 instance Swapped Either where
   swapped = iso (either Right Left) (either Right Left)
 
+instance (Swapped f, Swapped g) => Swapped (Product f g) where
+  swapped = iso f f
+    where
+      f (Pair x y) = Pair (x ^. swapped) (y ^. swapped)
+
+instance (Swapped p, Swapped q) => Swapped (Sum p q) where
+  swapped = iso f f
+    where
+      f (L2 x) = L2 (x ^. swapped)
+      f (R2 x) = R2 (x ^. swapped)
+
+instance (Swapped p) => Swapped (Flip p) where
+  swapped = iso f f
+    where
+      f (Flip p) = Flip (p ^. swapped)
+
+instance (f ~ g, Functor f, Swapped p) => Swapped (Biff p f g) where
+  swapped = iso f f
+    where
+      f (Biff p) = Biff (p ^. swapped)
+
+instance (Functor f, Swapped p) => Swapped (Tannen f p) where
+  swapped = iso f f
+    where
+      f (Tannen x) = Tannen $ fmap (^. swapped) x
+
+instance Swapped ((,,) x) where
+  swapped = iso f f
+    where
+      f ~(x,a,b) = (x,b,a)
```

Comment:
Why are these tuple instances using an irrefutable match? The instance for 2-tuples doesn't nor do any of the other new instances for product-types. If there's a good reason can we document it on the instances so people know to rely on it. If there isn't can we switch it to a normal pattern?

Reply:
I'm a proponent of lazy pattern matches for composability reasons, but I'm not tied to them. I figured it's a Kmett library who enjoys going full boar on these things. Considering `swapped` is a good candidate for lazy behavior, it seemed like a good idea.

Reply:
OK, then let's make them consistent and document them in the haddock comments